Code/Numerics/EigenSerializer/EigenBaseAddons.h

```diff
@@ -0,0 +1,32 @@
+// https://stackoverflow.com/questions/18382457/eigen-and-boostserialize
+
+#ifndef EIGEN_DENSE_BASE_ADDONS_H_
+#define EIGEN_DENSE_BASE_ADDONS_H_
+
+friend class boost::serialization::access;
+template <class Archive>
+void save(Archive& ar, const unsigned int version) const {
+  derived().eval();
+  const Index rows = derived().rows(), cols = derived().cols();
+  ar& rows;
+  ar& cols;
+  for (Index j = 0; j < cols; ++j)
+    for (Index i = 0; i < rows; ++i) ar& derived().coeff(i, j);
```

Comment:
In the RDKit code base we put braces around single line for and if blocks:
```suggestion
  for (Index j = 0; j < cols; ++j){
    for (Index i = 0; i < rows; ++i){
      ar& derived().coeff(i, j);
    }
  }  
```